akka-http-core/src/test/scala/akka/http/impl/engine/ws/WebSocketIntegrationSpec.scala

```diff
@@ -244,6 +245,57 @@ class WebSocketIntegrationSpec extends AkkaSpecWithMaterializer(
         resTry.failed.get.getMessage should ===("Connection failed.")
       }
     }
+
+    "fail the materialized future with akka.stream.StreamIdleTimeoutException if elements are not received within receive-idle-timeout" in {
+      val bindingFuture = Http().newServerAt("localhost", 0).bindSync({
+        _.attribute(webSocketUpgrade).get.handleMessages(Flow.apply, None)
```

Comment:
Flow.apply == Flow.identity so this is an echo-server, the test expects it to be a server that does not emit anything, something like `Flow.fromSinkAndSourceCoupled(Sink.ignore, Source.maybe[Message])` could work in a test (in real app you'd need to deal with text/binary, and also consume streaming but here we know it will be strict)